libnetwork/cni/network.go

```diff
@@ -81,7 +84,16 @@ func NewCNINetworkInterface(conf *InitConfig) (types.ContainerNetwork, error) {
 	// TODO: consider using a shared memory lock
 	lock, err := lockfile.GetLockfile(filepath.Join(conf.CNIConfigDir, "cni.lock"))
 	if err != nil {
-		return nil, err
+		// If we're on a read-only filesystem, there is no risk of
+		// contention. Fall back to a local lockfile.
+		if errors.Is(err, unix.EROFS) {
+			lock, err = lockfile.GetLockfile(filepath.Join(conf.RunDir, "cni.lock"))
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
```

Comment:
I'm probably overthinking this.  But if you establish a lock file in the RunDir due to the CNIConfigDir being readonly, and then that file in CNIConfigDir becomes writeable, could we end up with two lock files, and what does that mean?

Reply:
Future Podman invocations will use the new read-write lock, so everything should be fine.